aviate-labs/node-monitor
# 113-feature/record-all-node-providers-in-database

### Description
Fixes #113 

### Changes made
- add testing data to mimic response for the /node-provider endpoint from the ic api.
- add pydantic models for node provider(s).
- add table in database called `node_provider_lookup` which currently contains all node providers.
- add `update_node_provider_lookup_if_new()` to `node_monitor.py` which adds newly registered node providers to our database. This function will be called every day at 15:00 UTC

### Testing
- add CRUD tests for the `node_provider_lookup` table
- add tests to see is `update_node_provider_lookup_if_new()` reacts correctly when a new node provider is added to the api
- ran existing testing suite and all tests passed.
